src/foam/mlang/mlang.js

```diff
@@ -2757,10 +2757,16 @@ foam.CLASS({
       name: 'exprs',
       documentation: 'The expressions to be evaluated and returned in the projection. Typically are Properties.',
     },
+    {
+      class: 'Boolean',
+      name: 'buildObjectsFromProjectionWithClass',
+      documentation: 'if objects should be build from projectionWithClass (and they will not be retrieved with select)',
+      value: true
+    },
     {
       class: 'List',
       name: 'projectionWithClass',
-      documentation: 'The projection but with the class in position 0 with all other values offset by 1.',
+      documentation: 'The projection but with the class in position 1 and full object if requested at position 0 with all other values offset by 2.',
```

Comment:
We would only ever need the class or the object, but never both, so could we put either at position 0? Or, would that we awkward to deal with? We could just send an empty object, ie. {class: 'myClass'} with no properties set, which would be more consistent. I'm not saying we should, just thinking about the possibility.

Reply:
sure, thanks. fixed